libs/sdk-ui-filters/src/AttributeFilter/utils.ts

```diff
@@ -128,12 +128,21 @@ export function validateAttributeFilterProps(props: IAttributeFilterBaseProps) {
         "Hidden elements are not supported by the current backend implementation.",
     );
 
-    if (selectionMode === "single" && filter) {
-        invariant(
-            isPositiveAttributeFilter(filter) && attributeElementsCount(filterAttributeElements(filter)) < 2,
-            "Provided 'filter' property is not compatible with given selectionMode. It needs to be positive filter with max one item selected in attribute elements",
-        );
-    }
+    const isSingleSelect = selectionMode === "single";
+    const isMultiSelect = selectionMode !== "single";
+    const hasEmptyParentFilters = isEmpty(parentFilters);
+    const isPositiveWithMaxOneElement =
+        isPositiveAttributeFilter(filter) && attributeElementsCount(filterAttributeElements(filter)) < 2;
+
+    invariant(
+        isMultiSelect || !filter || (isSingleSelect && isPositiveWithMaxOneElement),
```

Comment:
Why something like `isSingleSelect ? isPositiveWithOneElement : true` is not sufficient?

Similar with other condition
`isSingleSelect ? hasEmptyParentFilters : true`

Or am I missing something?

Or if we want to go with nesting
```
if (isSingleSelect) {
    invariant(isPositiveWithOneElement, "...");

    invariant(hasEmptyParentFilters, "...");
}
```